### PR TITLE
Enable/disable tracking at runtime

### PR DESCRIPTION
**Pull request by GendelfLugansk**
This PR includes:

1. New ember imports

2. Ability to disable segment completely

3. Ability to disable/enable default page tracking and default identify calls at runtime

4. Ability to customize page tracking by defining `trackPageView` method in application controller

### CODE REVIEW

File: addon/services/segment.js [hunk 1]
```diff
@@ -1,88 +1,148 @@
...
   },
 
-  hasAnalytics: function() {
+  hasAnalytics: function () {
     return !!(window.analytics && typeof window.analytics === "object");
   },
 
-  // Default true unless user explicitly sets defaultPageTrack to false
-  pageTrackEnabled: function() {
+  enabled: function () {
```
Comment (josemarluedke) [inline addon/services/segment.js]: Do you mind naming this `isEnabled`?

File: addon/services/segment.js [hunk 2]
```diff
@@ -1,88 +1,148 @@
...
+    return !this.disabled();
+  },
+
+  disabled: function () {
```
Comment (josemarluedke) [inline addon/services/segment.js]: Do you mind naming this as `isDisabled` ?

File: addon/services/segment.js [hunk 3]
```diff
@@ -1,88 +1,148 @@
...
 
       this.log('trackPageView', arguments);
     }
   },
 
-  trackEvent: function(event, properties, options, callback) {
-    if(this.hasAnalytics()) {
+  trackEvent: function (event, properties, options, callback) {
+    if (this.enabled() && this.hasAnalytics()) {
+      this.checkPageTrackCalled();
```
Comment (josemarluedke) [inline addon/services/segment.js]: I believe this method call should not be in `trackEvent`.

File: config/ember-try.js [hunk 4]
```diff
@@ -1,5 +1,7 @@
 /* eslint-env node */
 module.exports = {
+  useYarn: true,
+  npmOptions: ['--ignore-engines'],
```
Comment (josemarluedke) [inline config/ember-try.js]: Why is this required?

**Comment by josemarluedke** [comment]
@sbrichards Do you mind talking a look in this PR as well?

File: config/ember-try.js (hunk 4 above)
Comment (GendelfLugansk) [reply config/ember-try.js]: To run on recent node versions. Ember-cli has `engines` key in package.json, in which they included only `"node": "^4.5 || 6.* || 7.*"`. Because of that yarn fails to install dependencies if node is newer than 7.

File: addon/services/segment.js (hunk 3 above)
Comment (GendelfLugansk) [reply addon/services/segment.js]: What method, `checkPageTrackCalled` ? It just logs a warning in console if `trackPageView` was never called. According to segment's documentation, some destinations require `page` call, so I added this.

**Comment by GendelfLugansk** [comment]
Tests are working when I run them with `ember try:each`. In travis output I see `[0712/193720.686481:FATAL:setuid_sandbox_host.cc(157)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I\'m aborting now. You need to make sure that /opt/google/chrome/chrome-sandbox is owned by root and has mode 4755.\n'`

Not sure how this could be fixed

File: addon/services/segment.js (hunk 2 above)
Comment (GendelfLugansk) [reply addon/services/segment.js]: I don't mind. Should I rename `pageTrackEnabled/pageTrackDisabled` as well or leave them as is?

**Comment by GendelfLugansk** [comment]
I found what caused previous build errors but now 2 jobs just timed out on `npm install` step

**Comment by josemarluedke** [comment]
Thank you @GendelfLugansk for your contributions.

**Comment by josemarluedke** [comment]
@GendelfLugansk Could you send a new PR adding tests for the features you added? Thank you.

**Comment by GendelfLugansk** [comment]
@josemarluedke this PR contains tests for 2 of 3 features. I'll add missing one.
